internal/template/template.go

```diff
@@ -128,6 +133,18 @@ func MergeData(dst map[string]interface{}, src map[string]interface{}) error {
 	return mergo.Merge(&dst, src, mergoOpts...)
 }
 
+// errTypeAssertion captures a CopyData error if deep copy cannot be asserted to be a map[string]interface{}
+var errTypeAssertion = errors.New("CopyData could not assert data type")
+
+// CopyData creates a deep copy of the argument map src.
+func CopyData(src map[string]interface{}) (map[string]interface{}, error) {
+	if data, ok := copier.Copy(src).(map[string]interface{}); ok {
```

Comment:
The more common pattern is to check if it is not ok, and return the error in that case. And you don't need to create a separate way for that. So in this case:

```
if data, ok := copier.Copy(src).(map[string]interface{}); !ok {
  return nil, errors.New("copyData could not assert data type")
}
return data, nil
```

And as a nit, in inner error messages, use lower case as the start of the sentence (e.g. https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/blob/master/internal/policygen/load.go#L57). It is because this error message can be appended to upper level error message and readability wise, having an upper case in it is not good. Usually only the out-most error message, for example, log.Errorf(...) in the top main function of the binary will have first latter capitalized as that is usually the start of the sentence.

See more at go/go-style/decisions#error-strings

Reply:
Also in non-test code, we normally don't directly mention the function name in the error message. Instead, interpret what the error is, or use "failed to ...". In this case, you can say something like "failed to assert output data type during data copy "